packages/fury-adapter-oas3-parser/test/unit/parser/oas/parseServersArray-test.js

```diff
@@ -0,0 +1,74 @@
+const { Fury } = require('fury');
+const { expect } = require('../../chai');
+const parse = require('../../../../lib/parser/oas/parseServersArray');
+const Context = require('../../../../lib/context');
+
+const { minim: namespace } = new Fury();
+
+describe('Servers Array', () => {
+  let context;
+
+  beforeEach(() => {
+    context = new Context(namespace);
+  });
+
+  it('warns when it is not an array', () => {
+    const servers = new namespace.elements.Object();
+
+    const parseResult = parse(context, servers);
+
+    expect(parseResult.length).to.equal(1);
+    expect(parseResult).to.contain.warning("'Servers Array' is not an array");
+  });
+
+  it('parses correctly when there is a single server', () => {
+    const server = new namespace.elements.Array([
+      {
+        url: 'https://user.server.com/1.0',
+        description: 'The production API server',
+      },
+    ]);
+
+    const parseResult = parse(context, server);
+
+    expect(parseResult.length).to.equal(1);
+
+    const servers = parseResult.get(0);
+
+    expect(servers).to.be.instanceof(namespace.elements.Array);
+    expect(servers.length).to.equal(1);
+    expect(servers.get(0)).to.be.instanceof(namespace.elements.Resource);
+    expect(servers.get(0).meta.content[0].content.value.content[0].content).to.equal('host');
+    expect(servers.get(0).meta.content[1].content.value.content).to.equal('The production API server');
+    expect(servers.get(0).attributes.content[0].content.value.content).to.equal('https://user.server.com/1.0');
```

Comment:
This check is very specific to the structure, I think it can be rewritten (including the following two lines) to use common accessors such as `classes`, `href`, `description`. I would try and avoid using `content` directly.

I'm not sure if this will work, but its atleast a clearer direction to try and refactor to:

```suggestion
    const host = servers.get(0);
    expect(host.classes.toValue()).to.deep.equal(['host']);
    expect(host.description.toValue()).to.equal('The production API server');
    expect(host.href.toValue()).to.equal('https://user.server.com/1.0');
```

Reply:
all replaced and working, I have to study the elements to shorten the structure during tests and functions.